src/components/confirmation-dialogs.js

```diff
@@ -104,3 +105,64 @@ export function confirmDocumentRevisionRestoration(t, revision, onOk, onCancel =
 
   dialog = confirm(createDialogProps());
 }
+
+export function confirmIdentityWithPassword({ t, username, userApiClient, onOk, onCancel = () => { } }) {
+  let dialog = null;
+  let createDialogProps = null;
+  let passwordValue = '';
+  let validationStatus = '';
+  let errorMessage = '';
+
+  const handleConfirmPassword = async password => {
+    try {
+      const { user } = await userApiClient.login({ username, password });
+      if (user) {
+        validationStatus = 'success';
+        errorMessage = '';
+        onOk();
+        return Promise.resolve();
```

Comment:
just `return;`, async functions already return a promise, no need to create an additional one.